kore/src/Kore/Internal/Substitution.hs

```diff
@@ -85,12 +93,69 @@ import Kore.TopBottom
 import Kore.Unparser
     ( unparseToString
     )
-import Kore.Variables.Fresh
-    ( FreshPartialOrd
-    )
 import Kore.Variables.UnifiedVariable
 import qualified SQL
 
+data Assignment variable =
+    Assignment
```

Comment:
I would prefer to give the pattern synonym a name matching the type, and the "real" constructor can have a funny name because it isn't exported:

```suggestion
    Assignment_
```